### DISCUSSION TITLE
Application Health Monitoring Ideas

### DISCUSSION THREAD

**Discussion by mike-weiner** [basecamp/kamal#1067] [Q&A] [answered]
Hi, folks. 

I've got a basic web application that I'm deploying with Kamal. I'd like to add some basic alerting to notify me if my application is down.

Is there a documented way that I could use Kamal for this? Or how are people achieving this today? 

My web application is basic. I don't need anything crazy complex. Bottom line is that I'd like to know about once per hour if my application is still running. 

Thank you!

**Comment by brodienguyen** [comment, marked as answer]
I am assuming you are talking about uptime monitoring. Something likes [PagerDuty](https://www.pagerduty.com/) or [Betterstack](https://betterstack.com/) would do what you are after. For self-host option, there is a one called [Uptime Kuma](https://github.com/louislam/uptime-kuma). Personally I am using [PagerDuty](https://www.pagerduty.com/) and [Betterstack](https://betterstack.com/) for production and [Uptime Kuma](https://github.com/louislam/uptime-kuma) for my internal homelab.

**Reply by mike-weiner** [reply]
@brodienguyen - Hi. Thanks for this. I'll need to do some more looking at these, but frankly these look to be more feature rich and complex than I need. 

My website displays weather station data. All I really need is a health check once a day to make sure things are running and if not I need a message that says "Hey - you need to check on your application... something is broken."

In my mind I was hoping for something like:

1. Kamal has health checks. Use that to run once a day to ensure containers are running.
2. Send result of health check to a webhook endpoint (i.e. Post result into Basecamp chat.)

But... I don't think that exists today. Wondering what the next closest option to that might be.

**Reply by sebastian** [reply]
Another option beyond those listed by @brodienguyen would be https://updown.io. It's incredibly simple, and incredibly cheap, and has been very reliable in the years I have used it. Can certainly recommend it. I believe it fits the bill quite nicely: checks a URL you give it at an interval you specify, and then sends you an email if something is off.

**Reply by mike-weiner** [reply]
Thanks for that suggestion, @sebastian! I took at look at that service. Frankly, I'm not interested in paying for a monitoring service for this side project of mine. It's just not _that_ important to me. I'm still exploring alternatives.

**Comment by mike-weiner** [comment]
I did end up going with BetterStack: https://betterstack.com/. Free. Simple. Basic enough for what I needed. 

I looked up Statuspage by Atlassain: https://www.atlassian.com/software/statuspage/features. It looked like another good option, but BetterStack had a few more features for their free plan.

**Comment by ThePatrykOOO** [comment]
For early-stage startups I’d prioritize:

1. 1-minute checks
2. SMS or phone alerts
3. Simple pricing (no credit systems)

You don’t need enterprise observability.

You need to know within 60 seconds if your API is down.

A [[lean uptime monitoring](https://alertsleep.com/uptime-monitoring/)](https://alertsleep.com/uptime-monitoring/) tool is usually enough until you scale.
